src/udev/udev-node.c

```diff
@@ -193,43 +197,58 @@ static int link_find_prioritized(sd_device *dev, bool add, const char *stackdir,
 
 static size_t escape_path(const char *src, char *dest, size_t size) {
         size_t i, j;
+        uint64_t h;
 
         assert(src);
         assert(dest);
 
         for (i = 0, j = 0; src[i] != '\0'; i++) {
                 if (src[i] == '/') {
-                        if (j+4 >= size) {
-                                j = 0;
-                                break;
-                        }
+                        if (j+4 >= size)
+                                goto toolong;
                         memcpy(&dest[j], "\\x2f", 4);
                         j += 4;
                 } else if (src[i] == '\\') {
-                        if (j+4 >= size) {
-                                j = 0;
-                                break;
-                        }
+                        if (j+4 >= size)
+                                goto toolong;
                         memcpy(&dest[j], "\\x5c", 4);
                         j += 4;
                 } else {
-                        if (j+1 >= size) {
-                                j = 0;
-                                break;
-                        }
+                        if (j+1 >= size)
+                                goto toolong;
                         dest[j] = src[i];
                         j++;
                 }
         }
         dest[j] = '\0';
         return j;
+
+toolong:
+        h = siphash24_string(src, UDEV_NODE_HASH_KEY.bytes);
+
+        assert(strlen(dest));
```

Comment:
this will access uninitialized memory, no? (since it will process a half-written, non-NUL-terminated string)

it would make sense to check the destination size here:

```
assert(size >= 12);
```

Reply:
Ouch!